SSINS/incoherent_noise_spectrum.py

```diff
@@ -633,8 +658,8 @@ def select(self, inplace=True, **kwargs):
 
         ins.metric_array.mask = np.copy(mask_uvf.flag_array)
         # In case this is called in the middle of the constructor.
-        if hasattr(ins, 'metric_ms'):
-            ins.metric_ms = ins.mean_subtract()
+        if not self._has_ins_data_params:
+            ins.set_ins_data_params()
 
```

Comment:
These two conditions are opposite to one another. I think I need to fiddle with some select-on-read test cases to see if it does what I want...There aren't any in the unit tests, which is something I will also add.

Reply:
maybe I screwed it up.